counting_sort/cpp/Counting_sort.cpp

```diff
@@ -0,0 +1,54 @@
+#include < iostream >
+  using namespace std;
+
+void countingSort(int arr[], int n, int RANGE) {
+  int count[RANGE] = {
+    0
+  };
+  int i;
+  int out[n];
+
+  for (i = 0; i < n; i++)
+    ++count[arr[i]];
+
+  for (i = 1; i < RANGE; i++)
+    count[i] += count[i - 1];
+
+  for (i = n - 1; i >= 0; i--) {
+    out[count[arr[i]] - 1] = arr[i];
+    --count[arr[i]];
```

Comment:
Do not use variable-length arrays.  Use an appropriately named ('k' followed by CamelCase) compile-time constant for the size.  [runtime/arrays] [1]

*Origin: CPPLintBear, Section: `all.cpplint`.*